deeposlandia/generator.py

```diff
@@ -5,54 +5,87 @@
+    """
+    reshaped_img = img.reshape([-1, 3])
+    label_img = np.zeros(shape=reshaped_img.shape[0], dtype=reshaped_img.dtype)
+    for label in label_config:
+        label_img[np.all(label['color']==reshaped_img, axis=1)] = label['id']
+    return label_img.reshape(img.shape[:3])
+
+
+def feature_detection_labelling(img, label_config):
     """One-hot encoding for feature detection problem
 
     Parameters
     ----------
     img : numpy.array
         Batched input image data of size (batch_size, image_size, image_size, 1)
-    label_ids : list of integer
-        Number of classes contained into the dataset
+    label_config : dict
+        Labels contained into the dataset
+
     Returns
     -------
     numpy.array
         Label encoding, array of shape (batch_size, nb_labels)
     """
+    if not (img.shape[1] == img.shape[2] and
+            (len(img.shape) == 3 or len(img.shape) == 4)):
+        raise ValueError(("Wrong image shape. Please provide batched images "
+                          "with equal width and height dimensions."))
+    label_ids = [item['id'] for item in label_config if item['is_evaluate']]
     if not all(isinstance(item, (int, np.uint8, np.uint32, np.uint64)) for item in label_ids):
         raise ValueError(("List of label IDs must contains "
                           "integers: {}").format(label_ids))
+    if len(img.shape) == 4 and img.shape[3] > 1:
+        img = recover_label_id(img, label_config);
     flattened_images = img.reshape(img.shape[0], -1).astype(np.uint8)
     one_hot_encoding = np.equal.outer(flattened_images, label_ids)
     return one_hot_encoding.any(axis=1)
 
 
-def semantic_segmentation_labelling(img, label_ids):
+def semantic_segmentation_labelling(img, label_config):
     """One-hot encoder for semantic segmentation problem
 
     Parameters
     ----------
     img : ndarray
         Batched input image data of size (batch_size, image_size, image_size, 1)
-    label_ids : list
-        List of dataset label IDs
+    label_config : dict
+        Label contained into the dataset
 
     Returns
     -------
     ndarray
         Label encoding, array of shape (batch_size, image_size, image_size, nb_labels), occurrence
     of the ith label for each pixel
     """
+    label_ids = [item['id'] for item in label_config if item['is_evaluate']]
     if not all(isinstance(item, (int, np.uint8, np.uint32, np.uint64)) for item in label_ids):
         raise ValueError(("List of label IDs must contains "
                           "integers: {}").format(label_ids))
+    if len(img.shape) == 4 and img.shape[3] > 1:
+        img = recover_label_id(img, label_config);
     img = img.squeeze().astype(np.uint8)
     one_hot_encoding = np.equal.outer(img, label_ids)
     return one_hot_encoding
 
 
-def feed_generator(datapath, gen_type, image_size, batch_size, seed=None):
+def feed_generator(datapath, gen_type, image_size, batch_size, col_mode, seed=None):
```

Comment:
do you have a checking or a `ValueError` if the `col_mode` doesn't have the right value ?

Reply:
I suggest `color_mode`. The `col_` in the variable name `col_mode` can mean "column", "color", etc.

Reply:
Done.